research/05-problem-ana.tex

```diff
@@ -4,27 +4,28 @@ \section{Problem Analysis}
 section. More specifically, this section describes the problems that
 are expected and have to be solved during the implementation process
 of the product. The first part of this section describes the process
-of incorporating REPL generation within the larger context of
+of incorporating a REPL within the larger context of
 Spoofax. The second part analyzes the part of the problem concerned
 with literate programming. This part will be analyzed in less detail
-than generating REPLs as it is unsure whether it will be present in
+than creating a REPL as it is unsure whether it will be present in
 the final product.
 
-\subsection{Incorporating REPL generation within Spoofax}
-\label{ssec:incorp-repl-gener}
-To generate properly working REPLs for any language defined in
-Spoofax, some Spoofax concepts have to be carefully considered. Every
-language can have many different language constructs. Generating
-REPLs, however, should not be constrained to any particular language
-constructs like classes, but should rather be done generically. For
-this reason the implementation and particularly the interaction with
-the user needs to be carefully considered. This sections lists some of
-the problems that are expected to arise from this.
+\subsection{Creating a REPL that works with any language defined in Spoofax}
+\label{ssec:incorp-repl-spoofax}
+To create a REPL that works with any language defined in Spoofax,
+some Spoofax concepts have to be carefully considered. Every language
+can have many different language constructs. The REPL for Spoofax,
+however, should not be constrained to any particular language
+constructs like classes, but should rather operate on any defined
+language. For this reason the implementation and particularly the
+interaction with the user needs to be carefully considered. This
+sections lists some of the problems that are expected to arise from
+this.
```

Comment:
I have openend an issue for this: https://github.com/spoofax-shell/doc/issues/33.